src/libraries/System.Private.Uri/src/System/DomainNameHelper.cs

```diff
@@ -351,19 +302,18 @@ private static unsafe bool IsIdnAce(char* input, int index)
 
                 if (!asciiLabel)
                 {
-                    string asciiForm = unescapedHostname.Substring(curPos, newPos - curPos);
                     try
                     {
-                        asciiForm = s_idnMapping.GetAscii(asciiForm);
+                        string asciiForm = s_idnMapping.GetAscii(hostname, curPos, newPos - curPos);
+
+                        dest.AppendString(s_idnMapping.GetUnicode(asciiForm));
```

Comment:
What's the purpose of this roundtripping through GetAscii and GetUnicode?  I see it's there before your change as well.